include/libsemigroups/fpsemi-examples.hpp

```diff
@@ -0,0 +1,539 @@
+  //! Chinese monoid, as described in
+  //! [10.1142/S0218196701000425](https://doi.org/10.1142/S0218196701000425).
+  //!
+  //! \param n the number of generators
+  //!
+  //! \returns A `std::vector<relation_type>`
+  //!
+  //! \noexcept
+  std::vector<relation_type> chinese_monoid(size_t n);
+
+  //! Construct a presentation for a monogenic semigroup.
+  //!
+  //! Returns a vector of relations giving a semigroup presentation defining the
+  //! monogenic semigroup defined by the presentation \f$\langle a \mid a^{m +
+  //! r} = a^m \rangle\f$.
+  //!
+  //! \param m the parameter `m`
+  //! \param r the parameter `r`
+  //!
+  //! \returns A `std::vector<relation_type>`
+  //!
+  //! \noexcept
+  std::vector<relation_type> monogenic_semigroup(size_t m, size_t r);
+
+  void
+  add_full_transformation_monoid_relations(std::vector<relation_type>& result,
+                                           size_t                      n,
+                                           size_t                      pi_start,
+                                           size_t e12_value);
+
+  template <typename T, typename F, typename... Args>
+  void setup(T& tc, size_t num_gens, F func, Args... args) {
+    tc.set_number_of_generators(num_gens);
+    for (auto const& w : func(args...)) {
+      tc.add_pair(w.first, w.second);
+    }
+  }
+
+  template <typename T>
+  T make_group(T& M) {
+    std::string              id = "e";
+    std::unordered_set<char> self_inverse;
+    self_inverse.emplace('e');
+
+    for (auto it = M.cbegin_rules(); it != M.cend_rules(); ++it) {
+      if (it->second == id && it->first.size() == 2
+          && it->first[0] == it->first[1]) {
+        // Remove self-inverse elements
+        self_inverse.emplace(it->first[0]);
+      } else if (it->first == id && it->second.size() == 2
+                 && it->second[0] == it->second[1]) {
+        // Remove self-inverse elements
+        self_inverse.emplace(it->second[0]);
+      }
+    }
+
+    std::string alphabet;
+    std::string inverse;
+
+    for (auto const& letter : self_inverse) {
+      alphabet += letter;
+      inverse += letter;
+    }
+
+    for (auto const& letter : M.alphabet()) {
+      if (self_inverse.find(letter) == self_inverse.cend()) {
+        alphabet += letter;
+        alphabet += std::toupper(letter);
+        inverse += std::toupper(letter);
+        inverse += letter;
+      }
+    }
+
+    T G;
+    G.set_alphabet(alphabet);
+    G.set_identity(id);
+    G.set_inverses(inverse);
+    for (auto it = M.cbegin_rules(); it < M.cend_rules(); ++it) {
+      if ((it->second != id || it->first.size() != 2
+           || it->first[0] != it->first[1])
+          && (it->first != id || it->second.size() != 2
+              || it->second[0] != it->second[1])) {
+        G.add_rule(*it);
+      }
+    }
+    return G;
+  }
+
+  template <typename T>
+  T AlternatingGroupMoore(size_t n) {
```

Comment:
This function should be renamed `alternating_group` and documented + behave like all the other functions in this file.